orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactResolver.java

```diff
@@ -120,6 +120,9 @@ public ResolveResult resolveExpectedArtifacts(Iterable<ExpectedArtifact> expecte
     ImmutableList.Builder<ExpectedArtifact> boundExpectedArtifacts = ImmutableList.builder();
 
     for (ExpectedArtifact expectedArtifact : expectedArtifacts) {
+      if (expectedArtifact == null) {
```

Comment:
eh? Why would an expected artifact be null? Is this due to clouddriver failing image resolution and deciding to return null instead of an error?

Reply:
It is not supposed to be null but looks like someone got NPE somehow(ref: https://github.com/spinnaker/spinnaker/issues/7004). We will investigate the reason but I believe allowing NPE in any case is bad, so this PR is to address those edge cases.